src/source/parserutil.d

```diff
@@ -101,3 +113,56 @@ void popMatchingDelimiter(alias openTokenType, Lexer)(ref Lexer lexer) {
 	assert(lexer.front.type == closeTokenType);
 	lexer.popFront();
 }
+
+unittest {
+	import source.context, source.dlexer;
+	auto context = new Context();
+
+	auto makeTestLexer(string s) {
+		import source.location;
+		auto base = context.registerMixin(Location.init, s ~ '\0');
+		return lex(base, context);
+	}
+
+	auto checkIncompleteInput(string s, string expectedMsg) {
+		auto lex = makeTestLexer(s).withComments(true);
+
+		import source.parserutil;
+		import source.exception;
+		import std.format;
+		lex.match(TokenType.Begin);
+
+		try {
+			throw unexpectedTokenError(lex, null);
+		} catch (IncompleteInputException e) {
+			assert(
+				e.msg == expectedMsg,
+				format!"Error message mismatch: expected `%s`, got `%s`"(
+					expectedMsg, e.msg)
+			);
+			return;
+		} catch (Throwable t) {
+			assert(
+				false,
+				format!"Didn't throw IncompleteInputException. Instead got %s"(
+					t)
+			);
+		}
+
+		assert(false, "Didn't throw IncompleteInputException");
+	}
+
+	// EOF in the middle of the token
+	checkIncompleteInput("/*/", "Comment must end with `*/`.");
+	checkIncompleteInput(
+		"\"", "Expected '\"' to end string literal, not the end of the file.");
+	checkIncompleteInput(
+		"'", "Expected a character literal, not the end of the file.");
```

Comment:
these are registered as mixins, so the end of the input.

Reply:
For some reason it passes on Windows and on linux when I test locally:
```
core.exception.AssertError@src/source/parserutil.d(138): Error message mismatch: expected `Expected '"' to end string literal, not the end of the input.`, got `Expected '"' to end string literal, not the end of the file.`
```

Reply:
You might xwant to delete /tmp..rdmd-NNNN where NNNN is your userid. Idk where the equivalent is on windows.

Sometime, dmd fucks things up and do not track dependencies properly.